src/components/page-layout/page-layout.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Sticky from 'react-stickynode';
+import debounce from 'debounce';
+
+class PageLayout extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      height: 0
+    };
+    this.debounceHandleWindowResize = debounce(() => {
+      const height = document.body.clientHeight;
+      this.setState({
+        bottomBoundaryValue: height - 450
+      });
+    }, 200);
+  }
+
+  componentDidMount() {
+    this.debounceHandleWindowResize();
+    window.addEventListener('resize', this.debounceHandleWindowResize);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.debounceHandleWindowResize);
+  }
+
+  render() {
+    const { props, state } = this;
+    return (
+      <div className="grid">
+        <div
+          className={`col col--4-mm col--12 ${
+            props.sidebarColor
+          } px0 mt0-mm mt60`}
```

Comment:
With reusable components, it's usually preferable to leave out layout styling that determines the component's relation to others around it. Instead, the component can be indifferent to where it's put — doesn't make assumptions about its context — and the parent can figure out how to arrange it in context.

Do you think that general rule applies here, or this is a worthy exception?

Reply:
I'm not sure about `worthy`, but here's the situation:

If `mt0-mm mt60` is removed, we'll need to figure out how to add it back or any page with a sticky top nav will look like this on mobile:

![introduction___mapbox](https://user-images.githubusercontent.com/10479155/42715786-4e297f7c-86ad-11e8-82ca-bae5de5bc489.png)

instead of this:

![introduction___mapbox](https://user-images.githubusercontent.com/10479155/42715817-7a22dfb0-86ad-11e8-86c3-05d19de266fa.png)

Does that seem to be a worthy exception @davidtheclark?

Reply:
Hmm, I don't think I understand. Also don't know about the `px0` class?

Reply:
I changed `sidebarColor` to `sidebarTheme` so it can include color and positioning. 🤷‍♀️